### PR TITLE
Add example with SolidStart

### PR DESCRIPTION
**Pull request by vitalikda**
# Description

Please include a summary of the change and which issue is fixed.
Please also include relevant motivation and context.

# Type of change

- [ ] 💔 Breaking
- [x] 🚀 Feature
- [ ] 🐛 Fix
- [ ] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [ ] 📖 Requires documentation update

### CODE REVIEW

File: examples/solid-start/src/utils/grafbase.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { GraphQLClient } from 'graphql-request'
+
+export const grafbase = new GraphQLClient('http://127.0.0.1:4000/graphql')
```
Comment (jvidalv) [inline examples/solid-start/src/utils/grafbase.ts:3]: Shouldn't this be a env variable?

Comment (jvidalv) [reply examples/solid-start/src/utils/grafbase.ts:3]: This should be an env variable and also configure the header x-api-key.

**Comment by jvidalv** [comment]
Hot reloading is not working after the app throws an error, is this expected?

Example, the x-api-key was missing in headers, the app errors: 

image

Then I update the file to include the x-api-key, but not even reloading the page it works again, I need to kill the vite process and start again.

**Comment by jvidalv** [comment]
It feels very unresponsive:

- Lack of loading states.
- No optimistic update.

File: examples/solid-start/src/utils/grafbase.ts (hunk 1 above)
Comment (vitalikda) [reply examples/solid-start/src/utils/grafbase.ts:3]: Right 👌

File: examples/solid-start/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+## Getting Started
+
+1. Run `npx degit grafbase/grafbase/examples/solid-start grafbase-with-solid` to clone this example
+2. [Deploy schema to Grafbase](https://grafbase.com/new/configure?template=Todo&source=https%3A%2F%2Fgithub.com%2Fgrafbase%2Fgrafbase%2Ftree%2Fmain%2Ftemplates%2Ftodo)
+3. Change directory into the new folder `cd grafbase-with-solid`
+4. Run `cp .env.example .env` to copy the example `.env.example` file to `.env`
+5. Open `.env` in your code editor, and provide your Grafbase API endpoint and API key
+6. Run `npm install`, or `yarn install` to install dependencies
+7. Run `npm run graphql`, or `yarn graphql` to generate types from Schema
+8. Run `vercel dev` and visit http://localhost:3000
```
Comment (jvidalv) [inline examples/solid-start/README.md:18]: You should update the readme

**Comment by jvidalv** [comment]
Amazing, now it feels way more responsive!

File: examples/solid-start/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+- [Grafbase](https://grafbase.com/) - learn about Grafbase features and API.
+
+To learn more about Solid, take a look at the following resources:
+
+- [SolidStart Documentation](https://start.solidjs.com/) - learn about Solid features and API.
+
+### Run on CodeSandbox
+
+[![Develop with CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://githubbox.com/grafbase/grafbase/tree/main/examples/nuxt)
```
Comment (notrab) [inline examples/solid-start/README.md:32]:
```suggestion
[![Develop with CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://githubbox.com/grafbase/grafbase/tree/main/examples/solid-start)
```

**Review by jvidalv — APPROVED** [review]
Here we go solid!

File: examples/solid-start/README.md (hunk 3 above)
Comment (vitalikda) [reply examples/solid-start/README.md:32]: Thanks!

**Pull request merged**